PayPal/src/main/java/com/braintreepayments/api/PayPalClient.java

```diff
@@ -18,19 +20,68 @@
 public class PayPalClient {
 
     private final BraintreeClient braintreeClient;
-    private final ApiClient apiClient;
-
     private final PayPalInternalClient internalPayPalClient;
 
+    private PayPalListener listener;
+
+    @VisibleForTesting
+    BrowserSwitchResult pendingBrowserSwitchResult;
```

Comment:
nit: we may be able to leverage shared prefs as a source of truth instead of member variables when notifying listeners of a pending browser switch result after `setListener()` is called.

Reply:
Actually talking it through with Sarah she reminded me about us needing to have access to the `Intent`. This is the best approach.